Automattic/wp-calypso
# Big Sky add health check for OpenAI

Related to #  https://github.com/Automattic/big-sky-plugin/issues/2299  

## Proposed Changes

* Adding health check for ai proxy before enabling the Build with AI button. 
* When AI is down, disable the button (not hiding the button). **This is up for discussion**.
* Currently it shows the button first, and only disables it once status check result is fetched. Alternatively, we can fetch the status first, then decide if to show the button, but it will cause a jump effect, which I don't enjoy.
<img width="978" alt="image" src="https://github.com/user-attachments/assets/5ce9022c-bb5d-4b98-b74a-7590572c39ff" />


## Why are these changes being made?


* So that we can fail gracefully when remote AI proxy is down and stop user from using it.

## Testing Instructions



* visit: http://calypso.localhost:3000/setup/site-setup/design-choices?siteSlug=YOUR_SITE_SLUG
* You can either sandbox public-api and force return value of `openai_api_healthcheck` to simulate proxy up/down.
* Or you can modify the return value of `checkHealth` to simulate.
* Then check if the button is enabled/disabled respectively.

## Pre-merge Checklist
